02_calculator/calculator.rb

```diff
@@ -15,18 +15,19 @@ def sum(array)
     return sum
 end
 
-def multiply(a, b)
-    return 0 if a == 0 || b == 0
-
-    return a * b
+def multiply(*numbers)
+    result = 1
+    numbers.each { |num| result *= num}
+    result
 end
 
 def power (a, b)
-    return 1 if b == 0
+    return a ** b
+end
 
-    result = a
-    (b - 1).times do 
-         sum *= a
-    end
-    return result
+def factorial(n)
+    return 1 if n == 0 || n == 1
```

Comment:
Nit: You don't need this guard clause either. `(2..0).each` and `(2..1).each` don't run anything inside the block, because 2 > 0 and 2 > 1.